pandas/core/internals/blocks.py

```diff
@@ -1627,7 +1627,10 @@ def _holder(self):
     @property
     def fill_value(self):
         # Used in reindex_indexer
-        return self.values.dtype.na_value
+        try:
```

Comment:
it might be worth instead adding this attribute as an alias on all EA dtypes, except of course Sparse which already defines this.

Reply:
Just checking I understand, this would involve the following:

* remove the try block from `ExtensionBlock.fill_value`
* add `ExtensionDType.fill_value` which just points to `ExtensionDType.na_value`

Is that right?

Reply:
I personally would not add this attribute to all dtypes. It really only makes sense for sparse (we also don't add a `tz` attribute for non-datetime dtypes, for example)

Reply:
Reason why I put this in `ExtensionBlock` is that `BlockManager` has references to `blk.fill_value` https://github.com/pandas-dev/pandas/blob/a45760fd45b434caf9107bb19f1536636cc3fbd8/pandas/core/internals/managers.py#L1262